### PR TITLE
[JENKINS-70220] Filter builds should be hidden when the big "no builds" block is visible

### PR DESCRIPTION
**Pull request by benebsiny**
See [JENKINS-70220](https://issues.jenkins.io/browse/JENKINS-70220).

The filter field is hidden when there's no build. Besides, when you input a keyword that doesn't match any build, the filter field is still displaying even if the "No builds" block appears. See demo.

Demo

### Testing done

The testing is added. But the testing doesn't cover the javascript code because the [Line 445](https://github.com/jenkinsci/jenkins/commit/7d2185827008378cb5492a66ed60935141facaae#diff-960d4968bf05bb1274736260bd76df827dad0bddd665a1fb6c3c628a8ef5d07bL445) and [Line 529](https://github.com/jenkinsci/jenkins/commit/7d2185827008378cb5492a66ed60935141facaae#diff-960d4968bf05bb1274736260bd76df827dad0bddd665a1fb6c3c628a8ef5d07bL529) will stop the javascript in testing mode. I'm not pretty sure if it is safe to remove the `isRunAsTest` or not so that I can add testing to cover the javascript.

### Proposed changelog entries

- Hides the filter field when there's no build in Build History Widget

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [X] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7673"></a>

### CODE REVIEW

**Review by daniel-beck — COMMENTED** [review]
Looks great.

Unsure whether a fairly narrow use case should result in a new attribute added to a generic (?) component. Could `clazz` be used instead, and rather than edit `style.display`, add or remove classes?

File: core/src/main/resources/lib/layout/search-bar.jelly
Comment (timja) [inline core/src/main/resources/lib/layout/search-bar.jelly:45]:
```suggestion
    <div class="jenkins-search ${attrs.clazz}">
```

File: core/src/main/resources/lib/layout/search-bar.jelly
Comment (timja) [inline core/src/main/resources/lib/layout/search-bar.jelly:38]:
```suggestion
```

File: war/src/main/js/filter-build-history.js
Comment (timja) [inline war/src/main/js/filter-build-history.js:45]: adjust this session to use the classList rather than inline style

**Review by timja — DISMISSED** [review]
please use existing visibility utils rather than inline styles

**Comment by benebsiny** [comment]
@timja Thanks for your feedback. JS and testing are modified.

**Comment by timja** [comment]
> @timja Thanks for your feedback. JS and testing are modified.

Thanks, code looks good, no time to test manually right now though

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Comment by NotMyFault** [comment]
Hey @benebsiny,

could you resolve the merge conflict, please?

**Comment by benebsiny** [comment]
@NotMyFault Hi, I've merged it, have a look at it please.

**Review by janfaracik — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by daniel-beck — APPROVED** [review]
Looks and works great in my testing. Thanks @benebsiny, and @timja for the assistance.

**Pull request merged**
